op-batcher/batcher/espresso.go

```diff
@@ -253,8 +257,94 @@ func (l *BlockLoader) EnqueueBlocks(ctx context.Context, blocksToQueue inclusive
 			continue
 		}
 
-		l.lastQueuedBlock = &blockRef
+		l.queuedBlocks = append(l.queuedBlocks, blockRef)
```

Comment:
Just to double-check my understanding, why do we replace the `lastQueuedBlock` with `queuedBlocks`? Is it to prevent unprocessed blocks from being dropped?

Reply:
We need to track the unsafe chain we're enqueuing to comprehensively detect reorgs, by analogy with how channel manager does it in original OP code. Without this we can't have checks on `firstQueuedBlock` in `nextBlockRange` for example